Content.Server/Pinpointer/NavMapSystem.cs

```diff
@@ -235,61 +254,83 @@ private void RefreshGrid(EntityUid uid, NavMapComponent component, MapGridCompon
         Dirty(uid, component);
     }
 
-    private void RefreshTileEntityContents(EntityUid uid, NavMapComponent component, MapGridComponent mapGrid, Vector2i chunkOrigin, Vector2i tile)
+    private NavMapChunk RefreshTileEntityContents(EntityUid uid, NavMapComponent component, MapGridComponent mapGrid, Vector2i chunkOrigin, Vector2i tile)
     {
         var relative = SharedMapSystem.GetChunkRelative(tile, ChunkSize);
         var flag = (ushort) GetFlag(relative);
         var invFlag = (ushort) ~flag;
+        var chunk = EnsureChunk(component, chunkOrigin);
 
         // Clear stale data from the tile across all entity associated chunks
         foreach (var category in EntityChunkTypes)
         {
-            if (!component.Chunks.TryGetValue((category, chunkOrigin), out var chunk))
-                chunk = new(chunkOrigin);
-
-            foreach (var (direction, _) in chunk.TileData)
-                chunk.TileData[direction] &= invFlag;
+            var data = chunk.EnsureType(category);
 
-            chunk.LastUpdate = _gameTiming.CurTick;
-            component.Chunks[(category, chunkOrigin)] = chunk;
+            foreach (var direction in data.Keys)
+            {
+                data[direction] &= invFlag;
+            }
         }
 
         // Update the tile data based on what entities are still anchored to the tile
         var enumerator = _mapSystem.GetAnchoredEntitiesEnumerator(uid, mapGrid, tile);
 
         while (enumerator.MoveNext(out var ent))
         {
-            if (!TryComp<AirtightComponent>(ent, out var entAirtight))
+            if (!_airtightQuery.TryComp(ent, out var airtight))
                 continue;
 
             var category = GetAssociatedEntityChunkType(ent.Value);
+            var data = chunk.EnsureType(category);
 
-            if (!component.Chunks.TryGetValue((category, chunkOrigin), out var chunk))
-                continue;
+            var directions = airtight.AirBlockedDirection;
 
-            foreach (var (direction, _) in chunk.TileData)
+            foreach (var direction in data.Keys)
             {
-                if ((direction & entAirtight.AirBlockedDirection) > 0)
-                    chunk.TileData[direction] |= flag;
+                if ((direction & directions) > 0)
```

Comment:
Is there a reason why this got split out into a variable? It looks a little harder to read.